File: lib/qt4/VersionListComboBox.cpp
/*
 * Copyright (c) 2013-2014 Nicholas Corgan (n.corgan@gmail.com)
 *
 * Distributed under the MIT License (MIT) (See accompanying file LICENSE.txt
 * or copy at http://opensource.org/licenses/MIT)
 */

#include <string>
#include <vector>

#include <pkmn/qt4/VersionListComboBox.hpp>
#include <pkmn/lists.hpp>

namespace pkmn
{
    namespace qt4
    {
        VersionListComboBox::VersionListComboBox(QWidget* parent):
            QComboBox(parent)
        {
            std::vector<pkstring> games_vec;
            get_version_list(games_vec);

            for(uint16_t i = 0; i < games_vec.size(); i++)
                addItem(QString::fromUtf16(games_vec[i]), QVariant(i));
        }
    } /* namespace qt4 */
} /* namespace pkmn */

File: lib/swig/java/ComboBoxes/VersionListComboBox.java
/*
 * Copyright (c) 2013-2015 Nicholas Corgan (n.corgan@gmail.com)
 *
 * Distributed under the MIT License (MIT) (See accompanying file LICENSE.txt
 * or copy at http://opensource.org/licenses/MIT)
 */

package nc.PKMN;

import javax.swing.*;

/**
 * A ComboBox that lists all versions.
 */
public class VersionListComboBox extends JComboBox<String>
{
    public VersionListComboBox()
    {
        super();

        StringVector versionVec = new StringVector();
        Lists.getVersionList(versionVec);
        for(int i = 0; i < versionVec.size(); i++)
        {
            addItem(versionVec.get(i));
        }

        setEditable(false);
    }
}
